crypto/ascon/ascon_aead128.c

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright 2025 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/*
+ * 64-bit optimised implementation of Ascon128 AEAD cipher.
+ *
+ * This file is based on LibAscon (CC0 1.0 license).
+ * Original authors: see LibAscon AUTHORS.md file
+ */
+
+# include "crypto/ascon.h"
+# include "ascon_internal.h"
+
+ASCON_API void
+ascon_aead128_encrypt(uint8_t* ciphertext,
+                      uint8_t* tag,

```

Comment:
Its unusual for an API in OpenSSL to separate the buffer lengths from the buffers.